src/cosmicqc/scdataframe.py

```diff
@@ -0,0 +1,164 @@
+"""
+Defines a SCDataFrame class for use in coSMicQC.
+"""
+
+import pathlib
+from typing import Any, Dict, TypeVar, Union
+
+import pandas as pd
+
+# provide backwards compatibility for Self type in earlier Python versions.
+# see: https://peps.python.org/pep-0484/#annotating-instance-and-class-methods
+SCDataFrame_type = TypeVar("SCDataFrame_type", bound="SCDataFrame")
+
+
+class SCDataFrame:
+    """
+    A class to handle and load different types of data files into a pandas DataFrame.
```

Comment:
For my understanding, why does this class need to exist for the CLI to work? What does this do differently or help with over just loading in the data as a data frame in the already existing functions?

I think this class is super simple and effective, just looking for the extra info that might be important to include in the documentation since I have found in other projects when they have their own object type, it lacks explanation why it is needed.

Reply:
Thanks great comment! It's generally useful because it reduces code duplication for the existing functions within this project found in the `analyze` module (for example, reading and writing files, or treating filepaths like dataframes) and felt strongly that there's an opportunity to do something useful with images inside the dataframe format + generate reports (like what's found in #39 ). I'll try to summarize this with some general thoughts in the comment here.